source/lambda/syncprocessor/lambda_function.py

```diff
@@ -126,12 +126,13 @@ def processImage(documentId, features, bucketName, outputBucketName, objectName,
 
     print("Processed Comprehend data for document: {}".format(documentId))
 
-    for key, val in opg_output[KVPAIRS].items():
-        if key not in comprehendAndMedicalEntities:
-            comprehendAndMedicalEntities[key] = val
-        else:
-            comprehendAndMedicalEntities[key].append(val)
-    opg.indexDocument(opg_output[DOCTEXT], comprehendAndMedicalEntities)
+    if 'ES_DOMAIN' in os.environ :
```

Comment:
Won't ES_DOMAIN be present in environ always? According to the definition in stack, we set it to undefined if the mode is kendra only. I don't think it removes the variable from environment.

Reply:
Answered below